### PR TITLE
update for existing imagestatus

### PR DESCRIPTION
**Pull request by sozercan**
Signed-off-by: Sertac Ozercan <sozercan@gmail.com>

**What this PR does / why we need it**:
Updates existing imageStatus objects

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/eraser/eraser.go [hunk 1]
```diff
@@ -169,8 +170,33 @@ func updateStatus(ctx context.Context, clientset *kubernetes.Clientset, results
...
+			}
+			_, err = clientset.RESTClient().Put().
+				AbsPath(apiPath).
+				Name(imageStatus.Name).
+				Resource("imagestatuses").
+				Body(body).DoRaw(ctx)
+			if err != nil {
+				log.Println("Could not update imagestatus for node: ", os.Getenv("NODE_NAME"))
+				return err
+			}
```
Comment (cpuguy83) [inline pkg/eraser/eraser.go]:
```suggestion
			}
			return nil
```

File: pkg/eraser/eraser.go [hunk 2]
```diff
@@ -169,8 +170,33 @@ func updateStatus(ctx context.Context, clientset *kubernetes.Clientset, results
 		Name(imageStatus.Name).
 		Resource("imagestatuses").
 		Body(body).DoRaw(ctx)
-
```
Comment (cpuguy83) [inline pkg/eraser/eraser.go]: Maybe `if !apierrors.IsAlreadyExists(err) { return err }` here and de-indent the block below?

File: pkg/eraser/eraser.go [hunk 3]
```diff
@@ -170,10 +171,36 @@ func updateStatus(ctx context.Context, clientset *kubernetes.Clientset, results
 		Name(imageStatus.Name).
 		Resource("imagestatuses").
 		Body(body).DoRaw(ctx)
-	if err != nil {
+	if !apierrors.IsAlreadyExists(err) {
+		log.Error(err, "failed to create ImageStatus")
```
Comment (cpuguy83) [inline pkg/eraser/eraser.go:175]: This is going to log an error when err is nil.

Comment (sozercan) [reply pkg/eraser/eraser.go:175]: good catch, removed log

**Review by cpuguy83 — APPROVED** [review]
lgtm

**Pull request merged**